shapeshift/web
# feat: swap fox farming v4 for v5

## Description

V5 contract starting November 18th 9AM MT, this will prepare us for it and should automagically work once it's live

## Notice




## Pull Request Type


## Issue (if applicable)

N/A

## Risk



Ensure current opportunities are still visible, see testing steps

## Testing



- Refresh with a fresh account or an account without FOX farming staking
- Ensure you're still seeing V4 as a farming opportunity in DeFi earn

- Refresh with an account with V4 staking
- Ensure you're still seeing your opportunity in DeFi earn/overview

### Engineering



- ☝🏽 for regression testing

To test the actual V5 feature:

- Apply this diff:

```shell
diff --git a/src/components/StakingVaults/AllEarnOpportunities.tsx b/src/components/StakingVaults/AllEarnOpportunities.tsx
index 6b6270370..d4e7213cb 100644
--- a/src/components/StakingVaults/AllEarnOpportunities.tsx
+++ b/src/components/StakingVaults/AllEarnOpportunities.tsx
@@ -70,7 +70,7 @@ export const AllEarnOpportunities = () => {
   })
 
   const filteredRows = useMemo(() => {
-    return allRows.filter(e => bnOrZero(e.tvl).gte(50000))
+    return allRows
   }, [allRows])
 
   const handleClick = useCallback(
```
- Ensure you're seeing this (we currently filter out opportunities by TVL, so we won't be seeing it until it goes live):

<img width="1271" alt="image" src="https://user-images.githubusercontent.com/17035424/202305572-51ab8da2-a2c9-4bb5-9352-52002e761594.png">

### Operations



- ☝🏽 
## Screenshots (if applicable)

<img width="672" alt="image" src="https://user-images.githubusercontent.com/17035424/202304612-ff483c73-a692-4e3a-9184-6602860b3655.png">
<img width="1264" alt="image" src="https://user-images.githubusercontent.com/17035424/202306294-d5f87e1a-c6ab-47f0-a7be-e726e8d925d0.png">
